patches/minecraft/net/minecraft/block/BlockChorusFlower.java.patch

```diff
@@ -0,0 +1,19 @@
+--- ../src-base/minecraft/net/minecraft/block/BlockChorusFlower.java
++++ ../src-work/minecraft/net/minecraft/block/BlockChorusFlower.java
+@@ -52,7 +52,7 @@
+             {
+                 int i = ((Integer)p_180650_3_.func_177229_b(field_185607_a)).intValue();
+ 
+-                if (i < 5 && p_180650_4_.nextInt(1) == 0)
++                if (i < 5 &&  net.minecraftforge.common.ForgeHooks.onCropsGrowPre(p_180650_1_, blockpos, p_180650_3_, p_180650_4_.nextInt(1) == 0))
+                 {
+                     boolean flag = false;
+                     boolean flag1 = false;
+@@ -141,6 +141,7 @@
+                     {
+                         this.func_185605_c(p_180650_1_, p_180650_2_);
+                     }
++                    net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.BlockEvent.CropGrowEvent.Post(p_180650_1_, p_180650_2_, p_180650_3_, p_180650_1_.func_180495_p(p_180650_2_)));
```

Comment:
Thanks @LexManos I am only trying to understand your wishes more clearly.  "Could probably make this smaller" isn't a very clear instruction (the sentence contains two conditional clauses).  "If this can be made smaller by using a ForgeHook, please do so" would have been more instructive.  Thank you for your input and I will update the other classes in this manner as well.